### ISSUE TITLE
Inconsistent and (currently) incorrectly documented use of template variables in components.

### ISSUE THREAD

**Issue by 0x426** [primefaces/primeng#16976]
### Describe the bug

I've upgraded to latest version of PrimeNG ([v18.0.0](https://github.com/primefaces/primeng/releases/tag/18.0.0)) just yesterday and noticed, that half of my app broke.

I checked out the [Migration Guide](https://v18.primeng.org/guides/migration), if there were any changes regarding the usage of template variables from beta / RC candidates to [v18.0.0](https://github.com/primefaces/primeng/releases/tag/18.0.0) but I couldn't find any.

For example the [Drawer](https://v18.primeng.org/drawer#template) was using the template variables like (does not work):
```
<p-drawer>
    <ng-template pTemplate="header"></ng-template>
</p-drawer>
```

Now, as it's correctly documented the usage is (works):
```
<p-drawer>
    <ng-template #header></ng-template>
</p-drawer>
```

---

The [Inplace](https://v18.primeng.org/inplace) component is for example incorrectly documented, usage "before" and like documented (does not work):
```
<p-inplace>
    <ng-template pTemplate="display">
        @if (this.data()?.name) {
            {{ this.data().name }}
        } @else {
            Loading spinner here
        }
    </ng-template>
    <ng-template pTemplate="content" let-closeCallback="closeCallback">
        Content here
    </ng-template>
</p-inplace>
```

But works only like this (works):
```
<p-inplace>
    <ng-template #display>
        @if (this.data()?.name) {
            {{ this.data().name }}
        } @else {
            Loading spinner here
        }
    </ng-template>
    <ng-template #content let-closeCallback="closeCallback">
        Content here
    </ng-template>
</p-inplace>
```

This also breaks Angulars control flow syntax in the component if used with `pTemplate`.

---

Now it gets even weirder. The [Table](https://v18.primeng.org/table#template) component. Let's take the `emptymessage` template for example:

It worked before like (as currently documented; works):
```
<p-table>
    <ng-template pTemplate="emptymessage">
        Empty message  here
    </ng-template>
</p-table>
```

But works also like this (not documented; works):
```
<p-table>
    <ng-template #emptymessage>
        Empty message  here
    </ng-template>
</p-table>
```

Now if we take the `rowexpansion` template, which use was like (and currently works):
```
<p-table>
    <ng-template pTemplate="rowexpansion" let-data>
        Expanded data row here
    </ng-template>
</p-table>
```

But per [documentation](https://v18.primeng.org/table#expand) it should be like (and does not work):
```
<p-table>
    <ng-template #rowexpansion let-data>
        Expanded data row here
    </ng-template>
</p-table>
```

---

Same for the [Panel](https://v18.primeng.org/panel#template):
```
<!-- Works -->
<p-panel [toggleable]="true">
    <ng-template pTemplate="header">Prime NG</ng-template>
    Example content
</p-panel>
  
<!-- Does not work-->
<p-panel [toggleable]="true">
    <ng-template pTemplate="header">
        Prime
        @if (this.dataLoaded()) {
            NG
        }
    </ng-template>
    Example content
</p-panel>

<!-- Works -->
<p-panel [toggleable]="true">
    <ng-template #header>
        Prime
        @if (this.dataLoaded()) {
            NG
        }
    </ng-template>
    Example content
</p-panel>
```

---

One last example, the [Increment Button](https://v18.primeng.org/inputnumber#buttons), the documentation says (works):
```
<p-inputnumber [showButtons]="true">
    <ng-template pTemplate="incrementbuttonicon">
        Increment button icon here
    </ng-template>
    <ng-template pTemplate="decrementbuttonicon">
        Decrement button icon here
    </ng-template>
</p-inputnumber>
```

But also the "undocumented" version works too:
```
<p-inputnumber [showButtons]="true">
    <ng-template #incrementbuttonicon>
        Increment button icon here
    </ng-template>
    <ng-template #decrementbuttonicon>
        Decrement button icon here
    </ng-template>
</p-inputnumber>
```

Either I'm missing some point in the documentation, tried to find some changes in the changelogs (but couldn't find anyone) or the use is intended but the documentation / migration guide outdated.

It's quiet hard to find the used components in a application and try out which template variable syntax works.

Thanks in advance.

### Environment

Angular: v18.0.0
PrimeNG: [v18.0.0](https://github.com/primefaces/primeng/releases/tag/18.0.0)

### Angular version

18.0.0

### PrimeNG version

18.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v22.9.0

### Expected behavior

Unified / consistent usage of template variables across all components.

**Comment by fontenele** [comment]
similar problem here... if a have template with same variable name (#item) inside a component with have template with same name (#item) then have conflict and nothing works
```html
<p-megaMenu [model]="rootItems">
      <ng-template #start>
             <p-select [options]="selectData">
                  <ng-template let-menuitem #item>
                        <div>{{ menuitem.name }}</div>
                  </ng-template>
            </p-select>
      </ng-template>
</p-megaMenu>
```




old version (works on prime 17)
```html
<p-megaMenu [model]="rootItems">
      <ng-template pTemplate="start">
             <p-dropdown [options]="selectData">
                  <ng-template let-menuitem pTemplate="item">
                        <div>{{ menuitem.name }}</div>
                  </ng-template>
            </p-dropdown>
      </ng-template>
</p-megaMenu>
```

**Comment by fontenele** [comment]
another case
```html
<p-card>
    <p-drawer>
        <ng-template #header>
            Trying Drawer header but having Card header
        </ng-template>
    </p-drawer>
</p-card>
```

**Comment by mehmetcetin01140** [comment]
Fixed in 6ce7b868ae2e356ab0b46603d7ae5ea151f10b42

**Comment by 0x426** [comment]
@mehmetcetin01140 Awesome, thanks for your efforts, will check as soon as a new release is available.
